### PR TITLE
Use regex to search for users instead of mongo's text search.

### PR DESCRIPTION
**Pull request by illegalprime**
Mongo was only able to search for tokens like `michael`, this is how checkin does it.
Also includes a fix for returning form items without answers.

### CODE REVIEW

File: server/routes/api/graphql.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ import { User, IUser, IFormItem } from "../../schema";
 import { Branches, Tags, AllTags } from "../../branch";
 import { schema as types } from "./api.graphql.types";
 
+import escapeStringRegexp = require("escape-string-regexp");
```
Comment (petschekr) [inline server/routes/api/graphql.ts]: This is the old, deprecated way of importing modules. Please use `import * as escapeStringRegexp from "escape-string-regexp;`

File: server/routes/api/graphql.ts [hunk 2]
```diff
@@ -44,23 +46,25 @@ const resolvers: IResolver = {
 			return allUsers.map(userRecordToGraphql);
 		},
 		search_user: async (prev, args) => {
+			const queryRegExp = new RegExp(escapeStringRegexp(args.search.trim()), "i");
```
Comment (petschekr) [inline server/routes/api/graphql.ts]: Could searching by regex not be a feature:tm: instead of a bug?

File: server/routes/api/graphql.ts (hunk 1 above)
Comment (illegalprime) [reply server/routes/api/graphql.ts]: I get an error when I do that with ts, something about it's type file not being a module

File: server/routes/api/graphql.ts (hunk 2 above)
Comment (illegalprime) [reply server/routes/api/graphql.ts]: What do you mean? Since it's escaped it should be case insensitive non regex search

Comment (petschekr) [reply server/routes/api/graphql.ts]: I mean what's the point of escaping possible regex characters in the first place if they can be used to make search more powerful and don't seem to have any security downsides?

Comment (illegalprime) [reply server/routes/api/graphql.ts]: Oh, I just think it would be confusing for normal people to use. We can add a regex switch

**Comment by petschekr** [comment]
@illegalprime I believe I just fixed both of those problems. Let me know if it looks good to you.

**Pull request merged**